src/Passes/Source/Generators/LlvmPassesConfig.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Commandline/ConfigurationManager.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    class LlvmPassesConfiguration
+    {
+      public:
+        // Default constructor which sets the standard pipeline.
+        LlvmPassesConfiguration();
+
+        // Setup and pre-fabricated configurations
+
+        /// Setup function that registers the different LLVM passes available via LLVM component.
+        void setup(ConfigurationManager& config);
+
+        /// Static function creates a new configuration where all transformations/validation requirements
+        /// are disabled. Using this configuration is equivalent to
+        static LlvmPassesConfiguration disable();
+
+        // Configuration interpretation
+        //
+
+        /// Returns true if the configuration disables all effects of this component. The effect of this
+        /// function being true is that registered component should have no effect on transformation
+        /// and/or validation of the QIR.
+        bool isDisabled() const;
+
+        /// Returns true if the current configuration is equivalent to the default configuration.
+        bool isDefault() const;
+
+        // Flags and options
+        //
+
+        /// Whether or not the LLVM AlwaysInline pass should be added to the profile.
+        bool alwaysInline() const;
+
+        /// Whether or not the default LLVM pipeline is disabled.
+        bool disableDefaultPipeline() const;
+
+        std::string passPipeline() const;
+
+      private:
+        // Variables that enables or disables the adding of specific passes
+        //
+
+        bool        always_inline_{false};            ///< Whether or not LLVM component should inline.
+        bool        disable_default_pipeline_{false}; ///< Whether or not the default pipeline is disabled
```

Comment:
The name `disable_default_pipeline_` tells me "The default pipeline should be disabled _at some time in the future_, but currently the default pipeline _is not necessarily disabled_". 
But the comment "Whether or not the default pipeline is disabled" tells me that the variable is a flag indicating whether the default pipeline _is already disabled_.

I.e. I feel a discrepancy between the variable name and the comment. Consider 
either renaming the variable to something like `default_pipeline_is_disabled_` 
or changing the comment to something like "Whether or not to disable the default pipeline".

Reply:
Changed.